src/software/new_geom/util/intersects.cpp

```diff
@@ -1,25 +1,119 @@
 #include "software/new_geom/util/intersects.h"
 
-bool intersects(const Polygon& poly, const Segment& segment)
+#include <limits>
+
+#include "software/new_geom/util/contains.h"
+#include "software/new_geom/util/distance.h"
+#include "software/new_geom/util/intersection.h"
+
+bool intersectsNew(const Polygon &first, const Segment &second)
```

Comment:
"New" added to the end of the function name to avoid some build issue, will revert with PR replacing the old `intersects`